processor/k8sattributesprocessor/internal/kube/kube.go

```diff
@@ -34,11 +34,37 @@ const (
 	// MetadataFromPod is used to specify to extract metadata/labels/annotations from pod
 	MetadataFromPod = "pod"
 	// MetadataFromNamespace is used to specify to extract metadata/labels/annotations from namespace
-	MetadataFromNamespace = "namespace"
+	MetadataFromNamespace  = "namespace"
+	PodIdentifierMaxLength = 16
```

Comment:
I think it can be even less to reduce memory footprint. Probably 4? It looks like the maximum number of resource attributes that we currently support https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8465/commits/3f4811e94f1dc150edda717a81b01fe30f3ace61#diff-da945276e27b1e0a3eed6a7e4a97c5b1bb9d90da43bd8ef9fce8b533584d4292R486-R494

Reply:
I agree, I would say that 2 is enough but set it to 4 just in case